app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -586,35 +586,37 @@ def event_message(mw_entity)
     private_constant :NotificationArgs
 
     def emit_middleware_notification(notification_args)
-      mw_entity = notification_args.entity_klass.find_by(:ems_ref => ems_ref) unless notification_args.entity_klass == MiddlewareServer
-      mw_server = if mw_entity.nil?
-                    MiddlewareServer.find_by(:ems_ref => notification_args.target_resource) ||
-                      MiddlewareDomain.find_by(:ems_ref => notification_args.target_resource)
-                  else
-                    MiddlewareServer.find_by(:id => mw_entity.server_id)
-                  end
-
-      return unless mw_server
-
-      Notification.create(
-        :type => notification_args.type, :options => {
-          :op_name   => notification_args.operation_name,
-          :op_arg    => notification_args.operation_args || '',
-          :mw_server => "#{mw_server.name} (#{mw_server.feed})"
-        }
-      )
-
-      unless mw_entity
-        EmsEvent.add_queue(
-          'add', id,
-          :ems_id          => id,
-          :source          => 'HAWKULAR',
-          :timestamp       => Time.zone.now,
-          :event_type      => notification_args.event_type(mw_server),
-          :message         => notification_args.event_message(mw_server),
-          :middleware_ref  => mw_server.ems_ref,
-          :middleware_type => mw_server.class.name.demodulize
+      ActiveRecord::Base.connection_pool.with_connection do
```

Comment:
@jrafanie Please review.

Reply:
Make sense to me.  Previously, if the thread is used to checkout out a connection to do the notification but then persists to do other things, activerecord won't steal the connection back from that thread.  Any other similar requests that come into puma will consume the database connection pool.   This is exactly what I was seeing when analyzing the logs.

This looks like it should fix the problem, provided this long running thread does no other activerecord database "work" before or after this block.

Reply:
@cfcosta looks great.  I'm good with merging this and releasing it if that's the only way to reasonably test it.

Reply:
@jrafanie great! There's no other ActiveRecord work being done besides generating notifications, so this will fix the problem.